### PR TITLE
IGNITE-22803 Put cluster ID in CmgInitMessage

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-22803

### CODE REVIEW

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java [hunk 1]
```diff
@@ -45,7 +45,22 @@ public interface ClusterTag extends NetworkMessage, Serializable {
...
+     *
+     * @param msgFactory Message factory to instantiate builder.
+     * @param name Cluster name.
+     * @param clusterId Cluster ID.
+     * @return Cluster tag instance.
+     */
+    static ClusterTag clusterTag(CmgMessagesFactory msgFactory, String name, UUID clusterId) {
         return msgFactory.clusterTag()
                 .clusterName(name)
                 .clusterId(UUID.randomUUID())
```
Comment (Phillippko) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java:66]: clusterID is ignored. Strange, that IDEA doesn't grey out unused parameters

File: modules/client-handler/src/integrationTest/java/org/apache/ignite/client/handler/TestServer.java [hunk 2]
```diff
@@ -126,7 +127,7 @@ ClientHandlerModule start(TestInfo testInfo) {
                 mock(IgniteComputeInternal.class),
                 clusterService,
                 bootstrapFactory,
-                () -> CompletableFuture.completedFuture(ClusterTag.clusterTag(msgFactory, "Test Server")),
+                () -> CompletableFuture.completedFuture(ClusterTag.clusterTag(msgFactory, "Test Server", UUID.randomUUID())),
```
Comment (Phillippko) [inline modules/client-handler/src/integrationTest/java/org/apache/ignite/client/handler/TestServer.java:130]: randomClusterTag?

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java [hunk 3]
```diff
@@ -45,7 +45,22 @@ public interface ClusterTag extends NetworkMessage, Serializable {
      * @param name Cluster name.
      * @return Cluster tag instance.
      */
-    static ClusterTag clusterTag(CmgMessagesFactory msgFactory, String name) {
+    static ClusterTag randomClusterTag(CmgMessagesFactory msgFactory, String name) {
```
Comment (Phillippko) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java:48]: I assume random tag should be used only in tests, let's annotate it as TestOnly? 

If it's really only for tests, do we really need a separate method? Name is changed, so we have to change all test classes in any case))

File: modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/AbstractClusterStateStorageManagerTest.java [hunk 4]
```diff
@@ -74,7 +73,7 @@ void tearDown() {
      */
     @Test
     void testClusterState() {
-        ClusterTag clusterTag1 = clusterTag(msgFactory, "cluster");
+        ClusterTag clusterTag1 = ClusterTag.randomClusterTag(msgFactory, "cluster");
```
Comment (Phillippko) [inline modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/AbstractClusterStateStorageManagerTest.java:76]: static import?

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java (hunk 1 above)
Comment (rpuch) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java]: Good catch!

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java (hunk 3 above)
Comment (rpuch) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterTag.java:50]: I added the annotation.

I think it still makes sense to use a special method, just to reduce duplication

**Pull request merged**